components/channel_header_dropdown/menu_items/leave_channel/leave_channel.js

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+
+import {showLeavePrivateChannelModal} from 'actions/global_actions';
+import {Constants} from 'utils/constants';
+
+export default class LeaveChannel extends React.PureComponent {
+    static propTypes = {
+
+        /**
+         * Object with info about user
+         */
+        channel: PropTypes.object.isRequired,
+
+        /**
+         * Boolean whether the channel is default
+         */
+        isDefault: PropTypes.bool.isRequired,
+
+        /**
+         * Object with action creators
+         */
+        actions: PropTypes.shape({
+
+            /**
+             * Action creator to leave channel
+             */
+            leaveChannel: PropTypes.func.isRequired,
+        }).isRequired,
+    };
+
+    handleLeave = (e) => {
+        e.preventDefault();
+
+        const {
+            channel,
+            actions: {
+                leaveChannel,
+            },
+        } = this.props;
+
+        if (channel.type === Constants.PRIVATE_CHANNEL) {
+            showLeavePrivateChannelModal(channel);
+        } else {
+            leaveChannel(channel.id);
+        }
+    }
+
+    render() {
+        const {channel, isDefault} = this.props;
+
+        if (isDefault) {
+            return null;
+        }
+
+        if (channel.type === Constants.DM_CHANNEL) {
+            return null;
+        }
+
+        if (channel.type === Constants.GM_CHANNEL) {
+            return null;
+        }
```

Comment:
Is there a style guide for multiline if-statement?

Reply:
I don't think we have but in several places we're doing the [control statements](https://github.com/airbnb/javascript#control-statements) like in https://github.com/mattermost/mattermost-webapp/blob/master/components/post_view/show_more/show_more.jsx#L32-L39.
But it might be my preference only, not suggesting to change unless you're into it 😄

Reply:
I actually prefer it the way it is, easier to read to me :) but I'm also 0/5